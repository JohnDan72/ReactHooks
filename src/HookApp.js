import React, { useState } from 'react';
// import PropTypes from 'prop-types';
import { Col, Container, Content, CustomProvider, FlexboxGrid, Header } from 'rsuite';
import NavBarHeader from './components/NavBarHeader';
import CounterApp from './components/01-useState/CounterApp';
import { Route, BrowserRouter as Router, Routes } from 'react-router-dom';
import CounterWithCustomHook from './components/01-useState/CounterWithCustomHook';
import SimpleForm from './components/02-useEffect/SimpleForm';
import SimpleFormWithCustomHook from './components/02-useEffect/SimpleFormWithCustomHook';
import MultipleCustomHooks from './components/03-examples/MultipleCustomHooks';
import FocusScreen from './components/04-useRef/FocusScreen';
import RealExampleRef from './components/04-useRef/RealExampleRef';
import Layout from './components/05-layoutEffect/Layout';


const HookApp = () => {
    const [theme, setTheme] = useState('dark');

    return (
        <>
            <CustomProvider theme={theme}>
                <Container>
                    <Header className="mt-2">
                        <FlexboxGrid justify="center">
                            <FlexboxGrid.Item as={Col} xs={20}>
                                <NavBarHeader setTheme={setTheme} />
                            </FlexboxGrid.Item>
                        </FlexboxGrid>
                    </Header>
                    <Content className="mt-3">
                        <FlexboxGrid justify="center">
                            <FlexboxGrid.Item as={Col} xs={20}>
                                <Router>
                                    <Routes>
                                        <Route exact path="/" element={<h2>Home</h2>} />
                                        <Route exact path="/counter" element={<CounterApp />} />
                                        <Route exact path="/counterHook" element={<CounterWithCustomHook />} />
                                        <Route exact path="/useEffect" element={<SimpleForm />} />
                                        <Route exact path="/useEffectHook" element={<SimpleFormWithCustomHook />} />
                                        <Route exact path="/useFetch" element={<MultipleCustomHooks />} />
                                        <Route exact path="/useRef" element={<FocusScreen />} />
                                        <Route exact path="/useRefReal" element={<RealExampleRef />} />
                                        <Route exact path="/useLayoutEffect" element={<Layout />} />
                                    </Routes>
                                </Router>
                            </FlexboxGrid.Item>
                        </FlexboxGrid>
                    </Content>
                </Container>
            </CustomProvider>
        </>
    );
}


export default HookApp;