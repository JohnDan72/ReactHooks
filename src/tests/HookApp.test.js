import { shallow } from "enzyme"
import HookApp from "../HookApp"


describe('Pruebas HookApp', () => {
    const [theme, setTheme] = ['dark', jest.fn()];

    test('debe renderizar el component', () => {
        const wrapper = shallow(<HookApp setTheme={setTheme}/>);

        expect(wrapper).toMatchSnapshot();
    })
})