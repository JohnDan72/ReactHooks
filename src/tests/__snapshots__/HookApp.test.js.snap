// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas HookApp debe renderizar el component 1`] = `
<Container>
  <Header
    className="mt-2"
  >
    <FlexboxGrid
      justify="center"
    >
      <FlexboxGridItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "as": [Function],
              "className": [Function],
              "classPrefix": [Function],
              "lg": [Function],
              "lgHidden": [Function],
              "lgOffset": [Function],
              "lgPull": [Function],
              "lgPush": [Function],
              "md": [Function],
              "mdHidden": [Function],
              "mdOffset": [Function],
              "mdPull": [Function],
              "mdPush": [Function],
              "sm": [Function],
              "smHidden": [Function],
              "smOffset": [Function],
              "smPull": [Function],
              "smPush": [Function],
              "xs": [Function],
              "xsHidden": [Function],
              "xsOffset": [Function],
              "xsPull": [Function],
              "xsPush": [Function],
            },
            "render": [Function],
          }
        }
        lg={16}
        md={18}
        xs={20}
      >
        <NavBarHeader
          setTheme={[MockFunction]}
        />
      </FlexboxGridItem>
    </FlexboxGrid>
  </Header>
  <Content
    className="mt-3"
  >
    <FlexboxGrid
      justify="center"
    >
      <FlexboxGridItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "as": [Function],
              "className": [Function],
              "classPrefix": [Function],
              "lg": [Function],
              "lgHidden": [Function],
              "lgOffset": [Function],
              "lgPull": [Function],
              "lgPush": [Function],
              "md": [Function],
              "mdHidden": [Function],
              "mdOffset": [Function],
              "mdPull": [Function],
              "mdPush": [Function],
              "sm": [Function],
              "smHidden": [Function],
              "smOffset": [Function],
              "smPull": [Function],
              "smPush": [Function],
              "xs": [Function],
              "xsHidden": [Function],
              "xsOffset": [Function],
              "xsPull": [Function],
              "xsPush": [Function],
            },
            "render": [Function],
          }
        }
        lg={16}
        md={18}
        xs={20}
      >
        <Outlet />
      </FlexboxGridItem>
    </FlexboxGrid>
  </Content>
</Container>
`;
